geo-types/src/geometry/geometry_collection.rs

```diff
@@ -248,78 +246,101 @@ impl<'a, T: CoordNum> GeometryCollection<T> {
+        /// let b = GeometryCollection::new_from(vec![point![x: 1.0, y: 2.01].into()]);
+        ///
+        /// approx::assert_relative_eq!(a, b, max_relative=0.1);
+        /// approx::assert_relative_ne!(a, b, max_relative=0.0001);
+        /// ```
+        #[inline]
+        fn relative_eq(
+            &self,
+            other: &Self,
+            epsilon: Self::Epsilon,
+            max_relative: Self::Epsilon,
+        ) -> bool {
+            if self.0.len() != other.0.len() {
+                return false;
+            }
+
+            self.iter()
+                .zip(other.iter())
+                .all(|(lhs, rhs)| lhs.relative_eq(rhs, epsilon, max_relative))
+        }
     }
-}
 
-#[cfg(any(feature = "approx", test))]
-impl<T> AbsDiffEq for GeometryCollection<T>
-where
-    T: AbsDiffEq<Epsilon = T> + CoordNum,
-    T::Epsilon: Copy,
-{
-    type Epsilon = T;
-
-    #[inline]
-    fn default_epsilon() -> Self::Epsilon {
-        T::default_epsilon()
+    impl<T> AbsDiffEq for GeometryCollection<T>
+    where
+        T: CoordNum + AbsDiffEq<Epsilon = T>,
+    {
+        type Epsilon = T;
+
+        #[inline]
+        fn default_epsilon() -> Self::Epsilon {
+            T::default_epsilon()
+        }
+
+        /// Equality assertion with an absolute limit.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// use geo_types::{GeometryCollection, point};
+        ///
+        /// let a = GeometryCollection::new_from(vec![point![x: 0.0, y: 0.0].into()]);
+        /// let b = GeometryCollection::new_from(vec![point![x: 0.0, y: 0.1].into()]);
+        ///
+        /// approx::abs_diff_eq!(a, b, epsilon=0.1);
+        /// approx::abs_diff_ne!(a, b, epsilon=0.001);
+        /// ```
+        #[inline]
+        fn abs_diff_eq(&self, other: &Self, epsilon: Self::Epsilon) -> bool {
+            if self.0.len() != other.0.len() {
+                return false;
+            }
+
+            self.into_iter()
+                .zip(other)
+                .all(|(lhs, rhs)| lhs.abs_diff_eq(rhs, epsilon))
+        }
     }
 
-    /// Equality assertion with an absolute limit.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use geo_types::{GeometryCollection, point};
-    ///
-    /// let a = GeometryCollection::new_from(vec![point![x: 0.0, y: 0.0].into()]);
-    /// let b = GeometryCollection::new_from(vec![point![x: 0.0, y: 0.1].into()]);
-    ///
-    /// approx::abs_diff_eq!(a, b, epsilon=0.1);
-    /// approx::abs_diff_ne!(a, b, epsilon=0.001);
-    /// ```
-    #[inline]
-    fn abs_diff_eq(&self, other: &Self, epsilon: Self::Epsilon) -> bool {
-        if self.0.len() != other.0.len() {
-            return false;
+    impl<T> UlpsEq for GeometryCollection<T>
+    where
+        T: CoordNum + UlpsEq<Epsilon = T>,
+    {
+        fn default_max_ulps() -> u32 {
+            T::default_max_ulps()
         }
 
-        let mut mp_zipper = self.into_iter().zip(other);
```

Comment:
I think this `mp_zipper` variable was copy/pasted from `MultiPolygon` or `MultiPoint`. No biggie, but I got rid of it since it's confusing in the context of `GeometryCollection`.